Userland/Libraries/LibJS/JIT/Compiler.h

```diff
@@ -27,6 +27,7 @@ class Compiler {
 #    if ARCH(X86_64)
     static constexpr auto GPR0 = Assembler::Reg::RAX;
     static constexpr auto GPR1 = Assembler::Reg::RCX;
+    static constexpr auto GPR2 = Assembler::Reg::R12;
```

Comment:
I dislike that this will always need an REX prefix to encode, even when staying in 32 bit land, but the other registers either have meaning, or may be lost between native calls, which can be bug prone

Reply:
Yeah, we could save 1 byte every time we use GPR2 if we choose another register, but it seems like a pretty low-value concern at this point. Something to keep in mind when we run out of ideas in the future ;)